extensions/reactive-messaging-http/runtime/src/main/java/io/quarkus/reactivemessaging/http/runtime/ReactiveHttpHandlerBean.java

```diff
@@ -0,0 +1,114 @@
+package io.quarkus.reactivemessaging.http.runtime;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.annotation.PostConstruct;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import org.eclipse.microprofile.reactive.messaging.Message;
+import org.jboss.logging.Logger;
+
+import io.quarkus.reactivemessaging.http.runtime.config.HttpStreamConfig;
+import io.quarkus.reactivemessaging.http.runtime.config.ReactiveHttpConfig;
+import io.smallrye.mutiny.Multi;
+import io.smallrye.mutiny.subscription.BackPressureStrategy;
+import io.smallrye.mutiny.subscription.MultiEmitter;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.ext.web.RoutingContext;
+
+/**
+ * a bean that handles incoming http requests
+ */
+@Singleton
+public class ReactiveHttpHandlerBean {
+
+    private static final Logger log = Logger.getLogger(ReactiveHttpHandlerBean.class);
+
+    @Inject
+    ReactiveHttpConfig config;
+
+    private final Map<String, Bundle<HttpMessage<?>>> httpProcessors = new HashMap<>();
+
+    @PostConstruct
+    void init() {
+        config.getHttpConfigs()
+                .forEach(this::addHttpProcessor);
+    }
+
+    void handle(RoutingContext event) {
+        String path = event.normalisedPath();
+        HttpMethod method = event.request().method();
+        String key = key(path, method);
+        Bundle<HttpMessage<?>> httpProcessorBundle = httpProcessors.get(key);
+        if (httpProcessorBundle != null) {
+            MultiEmitter<? super HttpMessage<?>> emitter = httpProcessorBundle.emitter;
+            StrictQueueSizeGuard guard = httpProcessorBundle.guard;
+            if (guard.prepareToEmit()) {
+                try {
+                    HttpMessage<Buffer> message = new HttpMessage<>(event.getBody(), new IncomingHttpMetadata(event.request()),
+                            () -> event.response().setStatusCode(202).end(),
+                            error -> {
+                                log.error("Failed to process message.", error);
+                                event.response().setStatusCode(500).end("Failed to process ");
+                            });
+                    emitter.emit(message);
+                } catch (Exception any) {
+                    guard.dequeue();
+
+                    log.error("Emitting message failed", any);
+                    event.response().setStatusCode(500).end();
```

Comment:
The rejected because of the overflow status I chose is 503 (line 65). After reading a bit, it seems to me that 429 is rather for situation when an API user exceeds a defined quota.